tests/dialects/test_cf.py

```diff
@@ -0,0 +1,31 @@
+from xdsl.ir import Block, OpResult
+from xdsl.dialects.arith import Addi, Subi, Muli, Constant
+from xdsl.dialects.builtin import i32
+from xdsl.dialects.cf import Branch, ConditionalBranch
+
+
+def test_branch():
+    a = Constant.from_int_and_width(1, i32)
+    b = Constant.from_int_and_width(2, i32)
+    # Operation to add these constants
+    c = Addi.get(a, b)
+
+    block0 = Block.from_ops([a, b, c])
+    _ = Branch.get(block0)
+
+
+def test_condbranch():
+    a = Constant.from_int_and_width(1, i32)
+    b = Constant.from_int_and_width(2, i32)
+    # Operation to add these constants
+    c = Addi.get(a, b)
+    d = Subi.get(a, b)
+    e = Muli.get(a, b)
+
+    block0 = Block.from_ops([a, b, c])
+    block1 = Block.from_ops([d])
+
+    branch0 = ConditionalBranch.get(c, block0, [d], block1, [e])
+    assert type(branch0.then) is OpResult
+    assert type(branch0.then_arguments[0]) is OpResult
+    assert type(branch0.else_arguments[0]) is OpResult
```

Comment:
Maybe also test that the right block is on the right side?

I think the operation you are building here is not actually valid. There are block arguments for blocks that do not take arguments. I guess what this test shows is that our verifiers are not strong enough for the conditional branch. 

Maybe construct the blocks with block arguments or remove them form the conditional branch op...

Reply:
> Maybe also test that the right block is on the right side?
Could you elaborate?

Reply:
You could do something like `assert branch0.then is block0`.

Reply:
THis is still an OpResult, no, I managed to access through successors, is that right?

Reply:
Yes, you are right. Sorry, I was confused due to the naming of `then`. I feel this should be called `condition` or something.